system/Database/BaseBuilder.php

```diff
@@ -2051,6 +2067,72 @@ public function onConstraint($set)
         return $this;
     }
 
+    /**
+     * Sets data source as a query for insert/update/upsert
+     *
+     * @param BaseBuilder|RawSql|string $query
```

Comment:
QueryBuilder does not accept raw SQL query.
Why don't we drop `string`?

Reply:
I added RawSql as a just in case scenario. 

Everything gets converted to a string so why not accept a string?

Reply:
I removed the string type.

Reply:
Can we also throw an exception when the variable is of type string? I'm honestly concerned that this could be abused. Not a lot of people are reading the docs when something is "just working".

Reply:
@sclubricants Can you throw an exception when the variable is of type string?

Reply:
Done